### ISSUE TITLE
[feature request] Preview nodes in real size

### ISSUE THREAD

**Issue by Djoop** [tikzit/tikzit#103]
Hi, thank you for developing this application. I think it would be great to render directly the nodes to their real sizes, as it really can change a lot the figure. Naturally, this would make things a bit more complicated as the dimensions of a node can depend on its textual content, but also on the font size, misc. arguments such as width or maximum width etc. 

Do you think it would be possible ? Do you think it would be a good idea, or on the contrary you decided on purpose to keep it simple as it is right now ?

**Comment by 0xACE** [comment]
I was sort of tinkering with it #82 but I haven't been near a full blown desktop in a while so that project is stalled for now...

**Comment by Djoop** [comment]
Thanks for your quick reply. I might try to have a look but I first need to get familiar with the code… might take some time. Have your modifications from #82 been released publicly on a fork or a different branch somewhere? It looks indeed that you tried to address multiples problems at the same time, not sure yet how everything is related, but I guess everything related to the rendering of the nodes and text would be useful (of course the rest as well, but for me it's not a big deal if e.g. the edges are not well rendered, as it will not change much the figure in the end, while the size and position of a node might drastically change the figure. Well it's at least how I see it, I guess different people make different figures using different functionalities…). Maybe also if some people prefer to "keep everything simple", we could have some option to choose between different rendering possibilities… not sure what is better yet.

**Comment by 0xACE** [comment]
> Have your modifications from #82 been released publicly on a fork or a different branch somewhere?

No. but I do have the intention to publish them once I clean up the commits, they are a mess in their current state (really bad mess, I worked on too many ideas at the same time without managing branches/commits nor respecting the current structure of the code)

> (of course the rest as well, but for me it's not a big deal if e.g. the edges are not well rendered, as it will not change much the figure in the end, while the size and position of a node might drastically change the figure. Well it's at least how I see it, I guess different people make different figures using different functionalities…)

The goal of #82  is to have `tikzit` render the preview window content very close to the final rendering so that you don't have to check between the windows all the time.

From my testing it seems like it only works well when you are sticking to "regular"/"normal" diagrams, as soon as you try to do fancy stuff the preview will be misaligned

My branch currently has these features (that I can remember):

- Fixed some segfaults (Don't even remember if this is true, but I think I did)
- Edge label support
- Multiline text edit box
- Nodes resize based on text
  - This works best when `tikzit` uses the same font as `latex` (requires that you run default settings/font)
  - Doesn't currently respect text size
  - Naively guesses you setup the node for multiline text so it respects `\newline` or `\\` (Casues some characters to display incorrectly when linebreaks appear next to another escape sequence `tikzit` recognizes)
  - Currently doesn't respect margin/padding
- Nodes respect one of those `max width` or w/e it was called but it's very incomplete...
- Edges are not exactly replicated but a "good enough" estimation
- Edges are not correctly placed on each shape (the polygons and what not, should have the edge be placed on the line but I just pretend it's an ellipses...)
- Labelplacement keybindings (a personal patch unlikely to be adopted `upstream`)
- Mousewheel bindings to select node/edge style (a personal patch unlikely to be adopted `upstream`)
- Keybindings for node/edge styles  (a personal patch unlikely to be adopted `upstream`)
- Create edges from selected or last created node (usable for creating diagrams fast) (a personal patch unlikely to be adopted `upstream`)
- Some more I can't remember but mostly personal patches

I would share some examples with you but again, I only have access to an old broken android phone...

Since it's difficult to epxlain everything in text you could share a `tiksstyle` and diagram and I could create a preview for you so you can see what my branch does...
